main.tf

```diff
@@ -1,10 +1,179 @@
+locals {
+  name                 = "ondat"
+  service_account_name = "storageos-operator"
+
+  ondat_etcd_endpoints = length(var.etcd_endpoints) == 0 ? "storageos-etcd.storageos-etcd:2379" : join(",", var.etcd_endpoints)
+
+  argocd_gitops_config = {
+    enable                             = true
+    etcdClusterCreate                  = length(var.etcd_endpoints) == 0
+    serviceAccountName                 = local.service_account_name
+    clusterSecretRefName               = "storageos-api"
+    clusterAdminUsername               = "storageos"
+    clusterAdminPassword               = "storageos"
+    clusterKvBackendAddress            = local.ondat_etcd_endpoints
+    clusterKvBackendTLSSecretName      = length(kubernetes_secret.etcd) > 0 ? kubernetes_secret.etcd[0].metadata[0].name : "storageos-etcd-secret"
+    clusterKvBackendTLSSecretNamespace = length(kubernetes_secret.etcd) > 0 ? kubernetes_secret.etcd[0].metadata[0].namespace : "storageos"
+    clusterNodeSelectorTermKey         = "storageos-node"
+    clusterNodeSelectorTermValue       = "1"
+    etcdNodeSelectorTermKey            = "storageos-etcd"
+    etcdNodeSelectorTermValue          = "1"
+  }
+
+  default_helm_values = [templatefile("${path.module}/values.yaml",
+    {
+      ondat_service_account_name   = local.service_account_name,
+      ondat_nodeselectorterm_key   = "storageos-node"
+      ondat_nodeselectorterm_value = "1"
+      etcd_nodeselectorterm_key    = "storageos-etcd"
+      etcd_nodeselectorterm_value  = "1"
+      ondat_admin_username         = "storageos",
+      ondat_admin_password         = "storageos",
+      ondat_credential_secret_name = "storageos-api",
+      etcd_address                 = local.ondat_etcd_endpoints,
+    },
+  )]
+}
+
 module "helm_addon" {
-  source               = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/kubernetes-addons/helm-addon?ref=v4.1.0"
-  manage_via_gitops    = var.manage_via_gitops
-  set_values           = local.set_values
-  set_sensitive_values = local.set_sensitive_values
-  helm_config          = local.helm_config
-  irsa_config          = local.irsa_config
-  addon_context        = var.addon_context
-  depends_on           = [kubernetes_storage_class.etcd, kubernetes_namespace.ondat]
+  source = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/kubernetes-addons/helm-addon?ref=v4.12.1"
+
+  manage_via_gitops = var.manage_via_gitops
+
+  helm_config = merge(
+    {
+      name             = local.name
+      chart            = "ondat"
+      repository       = "https://ondat.github.io/charts"
+      version          = "0.2.5"
+      namespace        = kubernetes_namespace.ondat.metadata[0].name
```

Comment:
this signals to Terraform that it needs to create the namespace first and avoids `depends_on`